### PR TITLE
Bug/ID-535/Some transitions between authentication activities take a long time

### LINKED ISSUE #288: Bug/ID-535/Some transitions between authentication activities take a long time
### Severity: Hight

**Linked Test Case Failed;**
https://app.testlodge.com/a/37307/projects/55109/suites/229241?expand_section=404319#case_4015981

**Description:**
The authentication activities or windows in the Hermes application exhibit poor performance, resulting in slow transitions between these activities. While other parts of the application perform efficiently, the specific actions related to authentication, such as logging in or signing up, experience significant delays, taking more than 8 seconds to complete compared to the expected maximum of 4 seconds.

**Steps to reproduce:**

1. Open the Hermes application. A screen with a button that says 'Sign Up with Google' will be displayed.
2. Click on the button that says 'Sign Up with Google'. You will be presented with a window with the Google accounts you have stored on your cell phone.
3. Select your Google account. Once selected you will be presented with a window with the map,
4. In the displayed tab click the button with the sandwich icon on the top left. A menu with different options will be displayed.
5. From the menu we will click on the 'User account' option.
Note that during each transition between views, there is a noticeable delay, significantly exceeding the expected response time.

**Expected result:**
Transitions between authentication activities should be smooth and fast, matching the performance of other activities in the application. The time required to switch between windows or activities should not exceed 5 seconds.

**Current result:**
Transitions between authentication activities exhibit poor performance, resulting in delays that are considerably longer than expected. These delays have a negative impact on the user experience and hinder the continuous flow of interactions within the application.

**Additional Information:**
* Reproducibility: This issue is reproducible consistently by following the steps mentioned above.  
* Slow transitions between authentication activities can lead to user frustration and a poor user experience.

**Attachments:**

Result:

Linked Logs:

### PR DESCRIPTION
**Pull request by JeferssonCL**

### CODE REVIEW

File: app/src/main/java/com/isc/hermes/model/User/UserRepositoryEditorUsingDBRunnable.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * and image path, to the database using an {@link AccountInfoManager}. It then retrieves the generated
+ * ID for the user's email and sets it in the user object. Finally, it updates the {@code UserRepository}
+ * with the created user using the {@link UserRepository#setUserContained} method.
+ */
+public class UserRepositoryEditorUsingDBRunnable implements Runnable {
+
+    private final User user;
+
+
+    public UserRepositoryEditorUsingDBRunnable(User user) {
```
Comment (aguirrekarina) [inline app/src/main/java/com/isc/hermes/model/User/UserRepositoryEditorUsingDBRunnable.java:31]: missing JavaDocs

File: app/src/main/java/com/isc/hermes/model/User/UserRepositoryEditorUsingDBRunnable.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        this.user = user;
+    }
+
+
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    @Override
+    public void run() {
```
Comment (aguirrekarina) [inline app/src/main/java/com/isc/hermes/model/User/UserRepositoryEditorUsingDBRunnable.java:43]: missing JavaDocs

File: app/src/main/java/com/isc/hermes/model/User/UserRepositoryEditorUsingDBRunnable.java (hunk 1 above)
Comment (JeferssonCL) [reply app/src/main/java/com/isc/hermes/model/User/UserRepositoryEditorUsingDBRunnable.java:31]: Done

File: app/src/main/java/com/isc/hermes/model/User/UserRepositoryEditorUsingDBRunnable.java (hunk 2 above)
Comment (JeferssonCL) [reply app/src/main/java/com/isc/hermes/model/User/UserRepositoryEditorUsingDBRunnable.java:43]: Done

**Review by aguirrekarina — APPROVED** [review]

**Review by CatCoder36 — APPROVED** [review]

**Review by VillalobosGarciaGaby — APPROVED** [review]

**Pull request merged**